engine/table/src/main/java/io/deephaven/engine/table/impl/MergedListener.java

```diff
@@ -69,19 +69,25 @@ public abstract class MergedListener extends LivenessArtifact implements Notific
     private Runnable delayedErrorReference;
 
     protected MergedListener(
-            Iterable<? extends ListenerRecorder> recorders,
-            Iterable<NotificationQueue.Dependency> dependencies,
-            String listenerDescription,
-            @Nullable QueryTable result) {
-        this.updateGraph = result != null
-                ? result.getUpdateGraph()
-                : ExecutionContext.getContext().getUpdateGraph();
+            final Iterable<? extends ListenerRecorder> recorders,
+            final Iterable<NotificationQueue.Dependency> dependencies,
+            final String listenerDescription,
+            final UpdateGraph updateGraph,
+            final @Nullable QueryTable result) {
+        this.updateGraph = updateGraph;
```

Comment:
I think we were wrong. We can always get the correct update graph from the context if we're constructing a listener.